### PR TITLE
Make Samsung backend work with SDK31

### PR DESCRIPTION
**Pull request by AhmedTElthakeb**
- Adding native library support 
- Updating backend settings

### CODE REVIEW

**Review by anhappdev — APPROVED** [review]

File: mobile_back_samsung/samsung/lib/sbe_config.hpp
Comment (d-uzlov) [inline mobile_back_samsung/samsung/lib/sbe_config.hpp:60]: `common_setting` are not used in UI.
It doesn't make sense to have `acceptable_value` for them because only the default value is used.
While `num_threads` is used in tflite-based backend, settings with IDs `configuration`, `share_results`, `cooldown` are not used anywhere in the project, they are dead weight. For Samsung backend `num_threads` is also likely a dead weight.
Recently we removed these settings to make backend configurations smaller and easier to read.
Please, revert changes that add `common_setting` values.

**Review by d-uzlov — CHANGES_REQUESTED** [review]

File: mobile_back_samsung/samsung/lib/sbe_config.hpp
Comment (AhmedTElthakeb) [reply mobile_back_samsung/samsung/lib/sbe_config.hpp]: Done

**Comment by AhmedTElthakeb** [comment]
When I test the app now, I see that image segmentation V2.0 is missing from evaluation (along with V0.7; so it ends up with only 4 BMs). I see there has been two relevant updates:
1) Remove Image Segmentation V0.7 (DeepLab V3) task (#401) [f216931fe78bf2ac775f467aa15b92e27f9f204a]
2) Include task file as an asset in the app (#394) [cdf27c197758d8e897a7d566b1bafd476fe0a415]

Any idea why this happens?

**Comment by anhappdev** [comment]
The CI test will fail if the number of benchmarks is not 5. Do you use custom task file?

**Comment by AhmedTElthakeb** [comment]
> The CI test will fail if the number of benchmarks is not 5. Do you use custom task file?

You mean under the dataset directory? I did not add one intentionally, may be from previous runs. What is the primary path for the task file now for flutter? I remember for android it was under assets ..

**Comment by anhappdev** [comment]
The default task file is bundled with the app 
https://github.com/mlcommons/mobile_app_open/blob/master/flutter/assets/tasks.pbtxt

Can you upload the APK you built here?

**Comment by AhmedTElthakeb** [comment]
Right I saw that, you mean this pointer?
https://github.com/mlcommons/mobile_app_open/blob/master/flutter/lib/resources/config_manager.dart#L14
Which points to this file
https://github.com/mlcommons/mobile_app_open/blob/master/flutter/assets/tasks.pbtxt

By bundled you mean this is part of the apk? so it is not downloaded to the phone?

I just emailed the APK to you (it is >25MB, cannot upload here)

**Comment by anhappdev** [comment]
Yes, the default task file is part of the app. You may have use a custom task file, see [custom-tasks.md](https://github.com/mlcommons/mobile_app_open/blob/master/flutter/documentation/custom-tasks.md).

I tested the APK you sent and on my test device with TFLite backend it shows all 5 benchmarks. Did you try delete the app and install it again?

**Comment by anhappdev** [comment]
You may also have used an outdated task file, see this bug https://github.com/mlcommons/mobile_app_open/issues/420

**Comment by anhappdev** [comment]
You need to make sure that you have defined all the `benchmark_setting` needed.
I did a quick check and see that there are only 3 entries for `IS_uint8_mosaic` in the file `mobile_back_samsung/samsung/lib/sbe_config.hpp` while there are 4 entries for `OD_uint8`.

**Review by d-uzlov — DISMISSED** [review]
_Just dismissing my previous review_

**Comment by AhmedTElthakeb** [comment]
> You need to make sure that you have defined all the `benchmark_setting` needed. I did a quick check and see that there are only 3 entries for `IS_uint8_mosaic` in the file `mobile_back_samsung/samsung/lib/sbe_config.hpp` while there are 4 entries for `OD_uint8`.

What entries are you referring to?

IS_uint8_mosaic flutter settings 
https://github.com/mlcommons/mobile_app_open/blob/3df403999e4d16635d0d712b7ac449bfc89c34d5/mobile_back_samsung/samsung/lib/sbe_config.hpp#L106-L142

OD_uint8 flutter settings 
https://github.com/mlcommons/mobile_app_open/blob/3df403999e4d16635d0d712b7ac449bfc89c34d5/mobile_back_samsung/samsung/lib/sbe_config.hpp#L144-L180

**Comment by anhappdev** [comment]
@AhmedTElthakeb: Your `sbe1200_config` has no `benchmark_setting` entry for `IS_uint8_mosaic`. So if your code uses the `sbe1200_config`, it will miss the v2.0 Image Segmentation benchmark.

**Comment by AhmedTElthakeb** [comment]
For flutter, sbe1200_config is not used.

**Comment by AhmedTElthakeb** [comment]
> When I test the app now, I see that image segmentation V2.0 is missing from evaluation (along with V0.7; so it ends up with only 4 BMs). I see there has been two relevant updates:
> 
> 1. Remove Image Segmentation V0.7 (DeepLab V3) task ([Remove Image Segmentation V0.7 (DeepLab V3) task #401](https://github.com/mlcommons/mobile_app_open/pull/401)) [[f216931](https://github.com/mlcommons/mobile_app_open/commit/f216931fe78bf2ac775f467aa15b92e27f9f204a)]
> 2. Include task file as an asset in the app ([Include task file as an asset in the app #394](https://github.com/mlcommons/mobile_app_open/pull/394)) [[cdf27c1](https://github.com/mlcommons/mobile_app_open/commit/cdf27c197758d8e897a7d566b1bafd476fe0a415)]
> 
> Any idea why this happens?

Issue is resolved!

**Review by d-uzlov — APPROVED** [review]

**Comment by anhappdev** [comment]
@AhmedTElthakeb I case you don't know, normally we let the creator of the PR merge it to preserve credit, so please merge the PR asap after approval to avoid merge conflict and effort to resolve it when the main code base changes.

**Pull request merged**
